File: hw3/Tests3/ProgramTests.fs
module Tests.ProgramTests
open Xunit
open hw3

[<Fact>]
let ``Main_WrongInputData_WillReturnNotZero`` () =
    let args = [|"1";"+";"&&&"|]
    let check = Program.main args
    Assert.True(check <> 0)

[<Fact>]
let ``Main_CorrectInputData_WillReturnZero`` () =
    let args = [|"1";"+";"3"|]
    let check = Program.main args
    let result = check = 0
    Assert.True(result)     

File: hw1/Tests/ProgramTests.cs
using hw1;
using Xunit;

namespace Tests
{
    public class ProgramTests
    {
        [Fact]
        public void Main_WrongInputData_WillReturnNotZero()
        {
            var args = new[] { "1", "+", "&&&" };
            var result = Program.Main(args);
            Assert.True(result != 0);
        }
        
        [Fact]
        public void Main_CorrectInputData_WillReturnZero()
        {
            var args = new[] { "1", "+", "3" };
            var result = Program.Main(args);
            Assert.True(result == 0);
        }
    }
}